common/templates/deployment.yaml

```diff
@@ -16,7 +16,10 @@ spec:
     {{- toYaml .Values.strategy | nindent 6 }}
   template:
     metadata:
-      {{- include "common.podAnnotations" . | nindent 6 }}
+      {{- if .Values.podAnnotations }}
+      annotations:
+        {{- include "common.podAnnotations" . | nindent 8 }}
```

Comment:
Maybe just move `checksum/config: {{ .Values.env | toString | sha256sum }}` to here